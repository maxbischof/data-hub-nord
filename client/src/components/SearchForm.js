import React from 'react'
import styled from 'styled-components'
import { ReactComponent as SearchIcon } from '../icons/search.svg'

export default function SearchForm({ setSearchTerm, resetSearchResults }) {
  let timeout = null

  function handleChange(event) {
    const searchTerm = event.target.value
    resetSearchResults('')
    clearTimeout(timeout)

    timeout = setTimeout(() => {
      setSearchTerm(searchTerm)
    }, 1000)
  }

  return (
    <Form onSubmit={(event) => event.preventDefault()}>
      <Label>
        <SearchIcon />
        <Input
          type="text"
          onChange={handleChange}
          placeholder="Suche nach Titel"
        />{' '}
      </Label>
    </Form>
  )
}

const Form = styled.form`
  border-radius: 20px;
  margin: 20px;
  align-self: center;
  padding: 5px;
  border: 1px solid var(--grey);
  width: 300px;
`

const Label = styled.label`
  display: flex;
  align-items: center;
`

const Input = styled.input`
  width: 100%;
  margin-left: 5px;
  border: hidden;
  &:focus {
    outline: none;
  }
`
