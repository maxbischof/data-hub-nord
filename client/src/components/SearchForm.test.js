import React from 'react'
import SearchForm from './SearchForm'
import renderer from 'react-test-renderer'

it('renders SearchForm component', () => {
  const renderedSearchForm = renderer.create(<SearchForm />).toJSON()
  expect(renderedSearchForm).toMatchSnapshot()
})
