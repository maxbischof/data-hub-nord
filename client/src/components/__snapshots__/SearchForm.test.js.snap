// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchForm component 1`] = `
<form
  className="sc-bdnylx jQOzXW"
>
  <label
    className="sc-gtssRu hAvyXn"
  >
    <svg>
      search.svg
    </svg>
    <input
      className="sc-dlnjPT geefGI"
      onChange={[Function]}
      type="text"
    />
     
  </label>
</form>
`;
